src/messaging.rs

```diff
@@ -167,4 +181,28 @@ impl<NodeUid: Ord> NetworkInfo<NodeUid> {
     pub fn num_faulty(&self) -> usize {
         self.num_faulty
     }
+
+    pub fn secret_key(&self) -> &SecretKey<Bls12> {
+        &self.secret_key
+    }
+
+    pub fn public_key_set(&self) -> &PublicKeySet<Bls12> {
+        &self.public_key_set
+    }
+
+    /// The canonical numbering of all nodes.
+    ///
+    /// FIXME: To avoid multiple computations of the same result, caching should be introduced.
+    pub fn node_indices(&self) -> BTreeMap<&NodeUid, u64> {
+        self.all_uids
+            .iter()
+            .enumerate()
+            .map(|(n, id)| (id, n as u64))
+            .collect()
+    }
+
+    /// Returns the unique ID of the Honey Badger invocation.
+    pub fn invocation_id(&self) -> Vec<u8> {
+        self.public_key_set.public_key().to_bytes()
+    }
```

Comment:
Maybe we should have another separate invocation ID; otherwise we make it unsafe to start two instances with the same key set.